xarray/tests/test_backends.py

```diff
@@ -992,6 +999,46 @@ def test_encoding_unlimited_dims(self):
             with pytest.warns(UserWarning):
                 ds.to_netcdf(tmp_file, engine='h5netcdf', unlimited_dims=['y'])
 
+class OpenMFDatasetTest(TestCase):
+    def validate_open_mfdataset_large_num_files(self, engine, nfiles=2000):
```

Comment:
We need to have a larger number of files opened than the limitation from the file system (> 1000) in order to demonstrate the problem.  I don't think we would have any problem for `nfiles=10` because this is much less than the open file limitation of the file system.